plans/add_compiler.pp

```diff
@@ -0,0 +1,68 @@
+# @summary Add a new compiler to a PE architecture or replace an existing one with new configuration.
+# @param avail_group_letter _ Either A or B; whichever of the two letter designations the compiler is being assigned to
+# @param compiler_fqdn _ The FQDN and certname of the new compiler
+# @param dns_alt_names _ A comma_separated list of DNS alt names for the compiler
+# @param primary_server_fqdn _ The FQDN and certname of the primary Puppet server
+# @param postgresql_server_fqdn _ The FQDN and certname of the PE-PostgreSQL server with availability group $avail_group_letter
+plan peadm::add_compiler(
+  String[1] $avail_group_letter,
+  Peadm::SingleTargetSpec $compiler_fqdn,
+  Optional[String[1]] $dns_alt_names = undef,
+  Peadm::SingleTargetSpec $primary_server_fqdn,
+  Peadm::SingleTargetSpec $postgresql_server_fqdn,
+){
```

Comment:
We should sanitize and create target objects here, similarly to how [other plans](https://github.com/puppetlabs/puppetlabs-peadm/blob/main/plans/action/install.pp#L54-L59) do it.

Reply:
Done.